targets/TARGET_STM/i2c_api.c

```diff
@@ -904,7 +904,8 @@ void HAL_I2C_ErrorCallback(I2C_HandleTypeDef *hi2c){
 
 #if DEVICE_I2CSLAVE
     /*  restore slave address */
-    i2c_slave_address(obj, 0, address, 0);
+    if(address != 0)
+        i2c_slave_address(obj, 0, address, 0);
```

Comment:
actually, one small remark, code may need small formatting update : space after if and use of braces
if (address != 0) {
    i2c_slave_address(obj, 0, address, 0);
}